### PR TITLE
Add video preview for assets

### PR DESCRIPTION
**Pull request by urbantrout**
This adds a simple preview for videos via HTML5 video player.

Related to #5564

### CODE REVIEW

File: src/assetpreviews/Video.php [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    public function getPreviewHtml(): string
+    {
+        /** @var Volume $volume */
+        $volume = $this->asset->getVolume();
+
+        if ($volume->hasUrls) {
+            $url = $this->asset->getUrl();
+        } else {
+            $source = $this->asset->getTransformSource();
+            $url = Craft::$app->getAssetManager()->getPublishedUrl($source, true);
```
Comment (brandonkelly) [inline src/assetpreviews/Video.php]: I don’t think we should be downloading video files just for previewing. Can you just throw a \yii\base\NotSupportedException` here instead? (I’ve updated the controller to be prepared for that possibility – 0319e770e6fa78f5140869a7aac955bcbd664398.)

Comment (urbantrout) [reply src/assetpreviews/Video.php]: That's a good idea!

**Comment by urbantrout** [comment]
I added `* @since 3.4.x` since I didn't know whether this would be merged nor when it would be merged,

**Comment by brandonkelly** [comment]
Merged in for 3.4.3 :)

**Pull request merged**